dyno-core/src/main/java/com/netflix/dyno/connectionpool/impl/lb/HostSelectionWithFallback.java

```diff
@@ -328,8 +333,29 @@ public void initWithHosts(Map<Host, HostConnectionPool<CL>> hPools) {
 		remoteDCNames.swapWithList(remoteDCSelectors.keySet());
 	}
 
+    /*package private*/ int calculateReplicationFactor(List<HostToken> allHostTokens) {
+        Map<Long, Integer> groups = new HashMap<>();
+
+        for (HostToken hostToken: allHostTokens) {
+            Long token = hostToken.getToken();
+            if (groups.containsKey(token)) {
+                int current = groups.get(token);
+                groups.put(token, current + 1);
+            } else {
+                groups.put(token, 1);
+            }
+        }
+
+        Set<Integer> uniqueCounts = new HashSet<>(groups.values());
+
+        if (uniqueCounts.size() > 1) {
+            throw new RuntimeException("Invalid configuration - replication factor cannot be asymmetric");
+        }
+
+        return uniqueCounts.toArray(new Integer[uniqueCounts.size()])[0];
```

Comment:
What happens that during an initialization, the 1st entry has fewer than the desired elements or count.